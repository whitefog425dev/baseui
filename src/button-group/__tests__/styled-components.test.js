/*
Copyright (c) 2018 Uber Technologies, Inc.

This source code is licensed under the MIT license found in the
LICENSE file in the root directory of this source tree.
*/
// @flow

import * as React from 'react';
import {shallow} from 'enzyme';

import {StyledRoot} from '../index.js';

describe('ButtonGroup styled components', () => {
  const styledComponents = [[StyledRoot, 'StyledRoot']];

  test.each(styledComponents)('default styled', (Component, name) => {
    const component = shallow(<Component />);
    expect(component.instance().getStyles()).toMatchSnapshot(
      `${name} has expected default styles`,
    );
  });

  test.each(styledComponents)('first', (Component, name) => {
    const component = shallow(<Component $first />);
    expect(component.instance().getStyles()).toMatchSnapshot(
      `${name} has expected styles when first element in group`,
    );
  });

  test.each(styledComponents)('last', (Component, name) => {
    const component = shallow(<Component $last />);
    expect(component.instance().getStyles()).toMatchSnapshot(
      `${name} has expected styles when last element in group`,
    );
  });
});
