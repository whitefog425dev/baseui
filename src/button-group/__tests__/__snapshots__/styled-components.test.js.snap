// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonGroup styled components default styled: StyledRoot has expected default styles 1`] = `
Object {
  "display": "flex",
}
`;

exports[`ButtonGroup styled components first: StyledRoot has expected styles when first element in group 1`] = `
Object {
  "display": "flex",
}
`;

exports[`ButtonGroup styled components last: StyledRoot has expected styles when last element in group 1`] = `
Object {
  "display": "flex",
}
`;
